### PR TITLE
Fix to module.run 

### PR DESCRIPTION
**Pull request by cachedout**
For @terminalmage to review

### CODE REVIEW

File: salt/states/module.py [hunk 1]
```diff
@@ -314,22 +314,31 @@ def _call_function(name, returner=None, **kwargs):
     :return:
     '''
     argspec = salt.utils.args.get_function_argspec(__salt__[name])
+
+    # func_kw is initialized to a dictinary of keyword arguments the function to be run accepts
     func_kw = dict(zip(argspec.args[-len(argspec.defaults or []):],  # pylint: disable=incompatible-py3-code
                    argspec.defaults or []))
+
```
Comment (s0undt3ch) [inline salt/states/module.py]: `s/keyword arguments/arguments`?

Comment (s0undt3ch) [reply salt/states/module.py]: Bah, the line bellow...

File: salt/states/module.py [hunk 2]
```diff
@@ -314,22 +314,31 @@ def _call_function(name, returner=None, **kwargs):
...
-                arg_type += funcset.values()
-                na_type.update(funcset)
-            else:
-                arg_type.append(funcset)
+            # We are just receiving a list of args to the function to be run, so just append
+            # those to the arg list that we will pass to the func.
+            arg_type.append(funcset)
         else:
-            func_kw.update(funcset)
+            for kwarg_key in funcset.keys():
```
Comment (s0undt3ch) [inline salt/states/module.py]: Do we really have to call keys?

File: salt/states/module.py (hunk 1 above)
Comment (cachedout) [reply salt/states/module.py]: Guh, yes. Will fix. Thanks.

File: salt/states/module.py (hunk 2 above)
Comment (cachedout) [reply salt/states/module.py]: No. I just was as explicit as I could be because this code is so dense and hard to read.

Comment (s0undt3ch) [reply salt/states/module.py]: But that does create a new list... :)

**Comment by rallytime** [comment]
@cachedout If this is merged, it replaces #43790, correct?

**Comment by cachedout** [comment]
Correct.

**Review by terminalmage — APPROVED** [review]
Mike has merged in some changes from me which make the tests fail in more useful ways, and also address @s0undt3ch's concern regarding iteration forming new lists on PY2.

**Review by thatch45 — APPROVED** [review]

**Pull request merged**